sqlglot/parser.py

```diff
@@ -2542,7 +2568,7 @@ def _parse_with(self, skip_with_token: bool = False) -> t.Optional[exp.With]:
             exp.With, comments=comments, expressions=expressions, recursive=recursive
         )
 
-    def _parse_cte(self) -> exp.CTE:
+    def _parse_cte(self) -> exp.Expression:
```

Comment:
So, regarding the type hint comment, this should ideally stay as `exp.CTE`.

Reply:
I tried to preserve it, but by using `try_parse` I was fighting with `make style` regarding this issue, will give it another shot